protocol/keepalive/serverclient_test.go

```diff
@@ -0,0 +1,151 @@
+package keepalive_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	ouroboros "github.com/blinklabs-io/gouroboros"
+	"github.com/blinklabs-io/gouroboros/protocol"
+	"github.com/blinklabs-io/gouroboros/protocol/keepalive"
+	ouroboros_mock "github.com/blinklabs-io/ouroboros-mock"
+)
+
+const (
+	// MockKeepAliveWrongCookie is the wrong cookie for a keep-alive response
+	MockKeepAliveWrongCookie uint16 = 1000
+)
+
+// ConversationEntryKeepAliveResponse is a pre-defined conversation entry for a keep-alive response
+var ConversationEntryKeepAliveResponseWrongResponse = ouroboros_mock.ConversationEntryOutput{
+	ProtocolId: keepalive.ProtocolId,
+	IsResponse: true,
+	Messages: []protocol.Message{
+		keepalive.NewMsgKeepAliveResponse(MockKeepAliveWrongCookie),
+	},
+}
+
+// ConversationKeepAlive is a pre-defined conversation with a NtN handshake and repeated keep-alive requests
+// and responses
+var ConversationKeepAliveWrongResponse = []ouroboros_mock.ConversationEntry{
+	ouroboros_mock.ConversationEntryHandshakeRequestGeneric,
+	ouroboros_mock.ConversationEntryHandshakeNtNResponse,
+	ouroboros_mock.ConversationEntryKeepAliveRequest,
+	ConversationEntryKeepAliveResponseWrongResponse,
+}
+
+func TestServerKeepaliveHandling(t *testing.T) {
```

Comment:
We should add a `defer` with goleak at the top of this test function to make sure all goroutines are cleaned up properly at the end